pmlc/dialect/pxa/analysis/strides.cc

```diff
@@ -213,22 +214,40 @@ StrideRange StrideInfo::range() const {
 }
 
 AffineValueExpr StrideInfo::toValueExpr(MLIRContext *ctx) const {
+  std::map<int64_t, std::pair<mlir::BlockArgument, int64_t>> ordered;
+
+  for (auto kvp : strides) {
+    int64_t loopDepth = 0;
+    auto block = kvp.first.getOwner();
+    auto parent = block->getParentOp();
+    while (!dyn_cast<FuncOp>(parent)) {
+      loopDepth++;
+      parent = parent->getParentOp();
+    }
+
+    ordered.emplace(
+        loopDepth * strides.size() + kvp.first.getArgNumber(),
```

Comment:
1) This code is incorrect, it is not guaranteed to give each index a unique number since the number of indexes in an affine.parallel is a) not the same from loop to loop in a nest, b) Not related to strides.size(), which could in fact be either bigger or smaller than the # of indexes.
2) Regardless, this mapping of two integers to 1 is a weird hacky thing to do anyway, just make a map of: 
std::map<std::pair<unsigned, unsigned>, std::pair<mlir::BlockArgument, int64_t>>
Pair is lexicographically ordered, so that should work just fine.  Also, the restriction to use int64_t everywhere is really only for things which might conceivable just to do arithmetic with indexes (like strides, sizes, etc), for things like loop depth or arg number, unsigned is fine.

Reply:
Whoops, didn't mean to approve!  But other than the comment inline, looks good.